pkg/network/protocols/events/consumer_test.go

```diff
@@ -78,6 +78,32 @@ func TestConsumer(t *testing.T) {
 	}
 }
 
+func TestInvalidBatchCountMetric(t *testing.T) {
+	kversion, err := kernel.HostVersion()
+	require.NoError(t, err)
+	if minVersion := kernel.VersionCode(4, 14, 0); kversion < minVersion {
+		t.Skipf("package not supported by kernels < %s", minVersion)
+	}
+
+	program, err := newEBPFProgram(config.New())
+	require.NoError(t, err)
+	defer program.Stop(manager.CleanAll)
+
+	ringBufferHandler := ddebpf.NewRingBufferHandler(1)
+	ringBufferHandler.RecordHandler(&ringbuf.Record{
+		RawSample: []byte("test"),
+	}, nil, nil)
+
+	consumer, err := NewConsumer("test", program, func([]uint64) {})
+	require.NoError(t, err)
+	consumer.handler = ringBufferHandler
+
+	consumer.Start()
+	consumer.Stop()
```

Comment:
the test is still flaky.
`consumer.Start` launches a go-routine, there's no guarantee it will run and process the event before hitting the `consumer.Stop`

Reply:
add Eventually to handle the issue